### PR TITLE
Add Go-Figure! recipe

### PR DESCRIPTION
**Pull request by hdetering**
Please add Go-Figure! build files to bioconda-recipes.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by hdetering** [comment]
Dear Bioconda team, is there anything I can do to help get this PR accepted?

**Review by mencian — CHANGES_REQUESTED** [review]
A comment in-line regarding SPDX identifiers, otherwise looks good to me!

File: recipes/go-figure/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    - adjusttext 0.7.3.1.*
+
+test:
+  commands:
+    - /usr/bin/env python --version
+    - python3 $(which gofigure.py) --help
+
+about:
+  home: https://gitlab.com/evogenlab/GO-Figure
+  license: GNU General Public License v3.0
```
Comment (mencian) [inline recipes/go-figure/meta.yaml:33]:
```suggestion
  license: GPL-3.0-or-later
```

File: recipes/go-figure/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+source:
+  url: https://gitlab.com/evogenlab/GO-Figure/-/archive/v{{ version }}/GO-Figure-v{{ version }}.tar.gz
+  sha256: 91728363c2fa1b1210355f665fdcfdd36287daa5fbaf05c2ac502c95fbb95b26
+  patches:
+    - change_data_path.patch
+
+build:
+  noarch: python
+  number: 0
```
Comment (mencian) [inline recipes/go-figure/meta.yaml:16]:
```suggestion
  number: 0
  run_exports:
    - {{ pin_subpackage('go-figure', max_pin="x") }}
```

**Review by mencian — APPROVED** [review]

**Pull request merged**